src/main/kotlin/com/cycode/plugin/components/toolWindow/components/treeView/components/detectionNodeContextMenu/DetectionNodeContextMenu.kt

```diff
@@ -99,26 +94,16 @@ class DetectionNodeContextMenu(
     }
 
     private fun onOpenInEditorOptionClicked() {
-        when (val node = getUnknownNode()) {
-            is SecretDetectionNode -> openSecretDetectionInFile(project, node)
-            is ScaDetectionNode -> openScaDetectionInFile(project, node)
-            is IacDetectionNode -> openIacDetectionInFile(project, node)
+        val node = getUnknownNode()
+        if (node is AbstractNode) {
+            openDetectionInFile(project, node)
         }
     }
 
     private fun onOpenViolationCardOptionClicked() {
-        when (val node = getUnknownNode()) {
-            is ScaDetectionNode -> {
-                treeView.displayScaViolationCard(node)
-            }
-
-            is SecretDetectionNode -> {
-                treeView.displaySecretViolationCard(node)
-            }
-
-            is IacDetectionNode -> {
-                treeView.displayIacViolationCard(node)
-            }
+        val node = getUnknownNode()
+        if (node is ScaDetectionNode) {
```

Comment:
I think we might want `AbstractNode` here? 

```suggestion
        if (node is AbstractNode) {
```

Reply:
thank you!!